src/main/scala/io/iohk/ethereum/BootstrapDownload.scala

```diff
@@ -0,0 +1,135 @@
+package io.iohk.ethereum
+
+
+import java.io.{File, FileInputStream, FileOutputStream}
+import java.net.URL
+import java.nio.file._
+import java.security.{DigestInputStream, MessageDigest}
+import java.util.zip.ZipInputStream
+
+import io.iohk.ethereum.utils.Logger
+
+
+/**
+  * A facility to
+  * - check the download location for a minimum amount of free space
+  * - download a zip from a URL and generate MD5 checksum
+  * - check the checksum
+  * - clean files out of given location
+  * - unzip to a given location
+  */
+object BootstrapDownload extends Logger {
+
+  val bufferSize = 4 * 1024
+  val leveldbFolderName = "leveldb"
+
+  private def assertAndLog(cond: Boolean, msg: String): Unit = {
+    if(!cond) log.info(msg)
+    assert(cond, msg)
+  }
+
+  def cleanOutFolder(pathToDownloadTo: Path): Unit = {
+    val leveldbFolder = pathToDownloadTo.toFile
+    assertAndLog(leveldbFolder.isDirectory, s"${pathToDownloadTo} must be a folder.")
+    assertAndLog(leveldbFolder.getName == leveldbFolderName, s"${pathToDownloadTo} must end in a folder named $leveldbFolderName")
+    leveldbFolder.listFiles().foreach(_.delete())
+  }
+
+  def downloadFile(urlToDownloadFrom: String, outFile: File): String = {
+
+    val md5 = MessageDigest.getInstance("MD5")
+    val is = new URL(urlToDownloadFrom).openStream()
+    try {
+      val dis = new DigestInputStream(is, md5)
+      try {
+        val out = new FileOutputStream(outFile)
+        try {
+          val buffer = new Array[Byte](bufferSize)
+          Stream.continually(dis.read(buffer)).takeWhile(_ != -1).foreach(out.write(buffer, 0, _))
+        } finally (out.close())
+        md5.digest.map("%02x".format(_)).mkString
```

Comment:
Same as `Hex.toHexString(md5.digest)`

Reply:
:+1: (Right, but the SO I cut and paste from didn't know that .... )